### PR TITLE
Add takeRecords() to PerformanceObserver

### PR DESCRIPTION
**Pull request by npm1**
@igrigorik 
@spanicker 
@tdresser

***
<a href="https://pr-preview.s3.amazonaws.com/npm1/performance-timeline/pull/97.html" title="Last updated on Nov 28, 2017, 7:24 PM GMT">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/performance-timeline/1aabb31...npm1:09ef81c.html" title="Last updated on Nov 28, 2017, 7:24 PM GMT">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -364,6 +365,11 @@ <h2><dfn>disconnect()</dfn> method</h2>
         <a>relevant global object</a>, and also empty <a>context object</a>'s
         <a>observer buffer</a>.</p>
       </section>
+      <section>
+        <h2><dfn>takeRecords()</dfn> method</h2>
+        <p>The <a>takeRecords()</a> method returns a copy of the <a>observer
+          buffer</a> and empties the observer buffer.</p>
```
Comment (tdresser) [inline index.html]: I think this needs to be the context object's observer buffer.

File: index.html [hunk 2]
```diff
@@ -364,6 +365,11 @@ <h2><dfn>disconnect()</dfn> method</h2>
         <a>relevant global object</a>, and also empty <a>context object</a>'s
         <a>observer buffer</a>.</p>
       </section>
+      <section>
+        <h2><dfn>takeRecords()</dfn> method</h2>
+        <p>The <a>takeRecords()</a> method returns a copy of the <a>observer
```
Comment (tdresser) [inline index.html]: I don't think the observer buffer can be directly returned.

Can we phrase this in terms of step 4.2?

Comment (npm1) [reply index.html]: Done. Looks almost equivalent to me

File: index.html (hunk 1 above)
Comment (npm1) [reply index.html]: Done

**Comment by npm1** [comment]
I created a W3C Account but after doing the confirmation I got on my email it still said it requires some sort of verification (maybe manually by someone in the group, not sure). If that happens then I can link that account to Github and the lpr check will stop failing I think

**Comment by spanicker** [comment]
Since Ilya will be back next week, makes sense to let him review.

**Comment by igrigorik** [comment]
Apologies about the delay. FWIW, I have an existing WIP PR that added takeRecords: https://github.com/w3c/performance-timeline/pull/95/commits/c29d56980ba470dcb5f0a6d012c3e5ad18eefb8c

My preference is to spell out the algorithm as in above PR. WDYT?

We can break apart the above PR and land takeRecords + other editorial updates in that PR, and move #entryTypes and buffer logic into separate PR.

**Comment by npm1** [comment]
The change from c29d569 looks good to me. Would you like to submit a PR to gh-pages with those changes? Or do you want me to modify this PR to describe takeRecords as  in c29d569?

**Comment by igrigorik** [comment]
Opened https://github.com/w3c/performance-timeline/pull/98, let's continue there.

**Pull request closed without merging**